src/wp-includes/rest-api/endpoints/class-wp-rest-terms-controller.php

```diff
@@ -154,10 +154,6 @@ public function register_routes() {
 	 * @return bool Whether the terms for the post can be read.
 	 */
 	public function check_read_terms_permission_for_post( $post, $request ) {
-		// If the requested post isn't associated with this taxonomy, deny access.
```

Comment:
This isn't a permissions check, it's a request validity check, so IMHO it should be moved out of `check_read_terms_permission_for_post`.